In [1]:
# import all the filenames
from allfunctions import *

1  def LassoCVModel(filename):

2  def OMPCVModel(filename):

3  def GradientBoostingCVModel(filename):

4  def RandomForestCVModel(filename):

5  def RidgeCVModel(filename):

6  def ElasticNetCVModel(filename):

7  def SVRPolyCVModel(filename):

8  def SVRSigmoidCVModel(filename):

9  def SVRLinearCVModel(filename):

10 def SVRRbfCVModel(filename):

In [4]:
def wrapper():
    if __name__ == '__main__':
        global model_selection
        global functionList
        global filenames
        
        functionList = [ LassoCVModel, GradientBoostingCVModel, 
                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel

        filenames = [ 'datasetIPPD.pickle' ]
        
        model_selection = {}
        
        for f in functionList:
            print()
            for filename in filenames:
                out = f(filename)
                newScore = out[filename]['train_rmse_cv_4_picking']

                if f in model_selection:
                    #compare scores and select more accurate model
                    oldScore = model_selection[f]['train_rmse_cv_4_picking']
                    if(newScore < oldScore):
                        print("【【【】】】switching models for %s..." % filename)
                        print("【【【】】】newScore: %f < oldScore: %f" % (newScore, oldScore))
                        print("【【【】】】newModel: %s \n" % f)
                        print(100*'_')
                        model_selection[f] = out[filename]
                else:
                    #no score to compare
                    print("【【】】adding new model for %s \n" % filename)
                    print(100*'_')
                    model_selection[f] = out[filename]
    return model_selection

In [5]:
t = wrapper()


Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.691380811147901}
Lasso(alpha=20.691380811147901, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for datasetIPPD.pickle 

____________________________________________________________________________________________________

Dataset size read: train 256 and test 65 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'learning_rate': 0.001, 'loss': 'ls'}
GradientBoostingRegressor(alpha=0.9, criterion='friedman_mse', init=None,
             learning_rate=0.001, loss='ls', max_depth=None,
             max_features=None, max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=1,
         

In [6]:
for key in t.keys():
    print(key)
    print(t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

<function LassoCVModel at 0x7f64208e7c80>
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
<function GradientBoostingCVModel at 0x7f63fa3b07b8>
421.108424701 342.356068455 1532.34615385
____________________________________________________________________________________________________
<function RandomForestCVModel at 0x7f63fa3b0840>
420.621205322 342.402960902 1532.34615385
____________________________________________________________________________________________________
<function RidgeCVModel at 0x7f63fa3b08c8>
1025.08583073 880.826144692 1532.34615385
____________________________________________________________________________________________________
<function ElasticNetCVModel at 0x7f63fa3b0950>
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
<function SVRPolyCVModel at 0x7f63fa3b09d8>
1032.957

# (4C4) Response when four of four modules are chosen as training dataset

'''
### LassoCVModel at 0x7f64208e7c80
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### GradientBoostingCVModel at 0x7f63fa3b07b8
421.108424701 342.356068455 1532.34615385
____________________________________________________________________________________________________
### RandomForestCVModel at 0x7f63fa3b0840
420.621205322 342.402960902 1532.34615385
____________________________________________________________________________________________________
### RidgeCVModel at 0x7f63fa3b08c8
1025.08583073 880.826144692 1532.34615385
____________________________________________________________________________________________________
### ElasticNetCVModel at 0x7f63fa3b0950
1025.08609776 880.826424842 1532.34615385
____________________________________________________________________________________________________
### SVRPolyCVModel at 0x7f63fa3b09d8
1032.95723148 882.068031346 1532.34615385
____________________________________________________________________________________________________
### SVRSigmoidCVModel at 0x7f63fa3b0a60
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRLinearCVModel at 0x7f63fa3b0ae8
1080.46665239 1059.84229153 1532.34615385
____________________________________________________________________________________________________
### SVRRbfCVModel at 0x7f63fa3b0b70
1080.46665239 1059.84229153 1532.34615385
'''

In [7]:
import pickle

with open('model_performance_all_modules.pickle', 'wb') as handle:
    pickle.dump(t, handle, protocol=pickle.HIGHEST_PROTOCOL)


In [3]:
for mem in t.keys():
    print(100*'_')
    print(t[mem])
    print(100*'_')

NameError: name 't' is not defined